price/priceService.ts

```diff
@@ -0,0 +1,21 @@
+const baseUrl:string = 'https://dydx-testnet.imperator.co/v4';
+
+function fetchCandles(pair:string,timeFrame:string):void{
+    fetch(`${baseUrl}/candles/perpetualMarkets/${pair}?resolution=${timeFrame}`)
+    .then(response => {
+        if(!response.ok){
+            throw new Error('Network response was not ok');
+        }
+        else{
+            return response.json();
+        }
+    })
+    .then(data=>{
+        console.log(data);
+    })
+    .catch(error=>{
```

Comment:
You don't catch the error. Plugins do.
A simple rule is that, you catch the error when you can handle it. As a _host_, we don't know how to essentially handle the error. Instead, the plugin developers exactly know that when an error occurs, what they should do with it.